pkg/stanza/fileconsumer/internal/reader/metadata.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package reader // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/reader"
+
+import (
+	"time"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/fileconsumer/internal/fingerprint"
+)
+
+type Metadata struct {
+	Fingerprint     *fingerprint.Fingerprint
+	Offset          int64
+	FileAttributes  map[string]any
+	HeaderFinalized bool
+
+	LastSplitInputChange time.Time
+	LastSplitInputLength int
+}
+
+// Getters and Setters for LastDataChange and LastDataLength
+func (m *Metadata) LastDataChange() time.Time {
+	return m.LastSplitInputChange
+}
+
+func (m *Metadata) SetLastDataChange(t time.Time) {
+	m.LastSplitInputChange = t
+}
+
+func (m *Metadata) LastDataLength() int {
+	return m.LastSplitInputLength
+}
+
+func (m *Metadata) SetLastDataLength(l int) {
+	m.LastSplitInputLength = l
```

Comment:
Why do we need to provide both setters/getters and keep the fields exported?

Reply:
The design is a little unusual and maybe could be better, but here's what factors in to this:
1. `reader.Metadata` is used to encode the state of each reader. Essentially it represents a rich checkpoint for a file. Therefore, these fields need to exported to be included in the encoding.
2. At the same time, I think the flushing functionality is not limited to the readers. Any split func could reasonably be wrapped with this flushing mechanism. Therefore, I'm trying to make the functionality independent of a given implementation. Since the flushing logic specifically works with these two fields, I though having Getters and Setters in an interface may communicate clearly the requirements without making the implementer do any logic themselves.

The other design which I considered is to export a `flush.State` struct and embed this in the `reader.Metadata`. The downside is that then the `reader.Metadata` encodes an externally struct. I'd rather it be as flat and simple as possible. After looking at this more, I think maybe it's the better approach after all though. WDYT?

Reply:
I think an embedded `flush.State` struct would be cleaner. Do you expect any performance implications from encoding it?

Also `Flusher` interface sounds a bit confusing. I would expect it to have `Flush` method. Can we replace the interface with `flush.State` struct so we don't need any setters/getters?

Reply:
I ran the benchmarks and did not see a notable difference due to encoding the additional fields. This is what I would expect because this additional encoding work is only once per file, per poll interval, so should be minimal compared to opening, reading, and tokenizing each file.

Reply:
I've updated this to use a `flush.State` struct instead of the interface.